src/background.ts

```diff
@@ -1339,26 +1417,34 @@ app.on("before-quit", (event) => {
   const killingProcessPromises = killEngineAll();
   const numLivingEngineProcess = Object.entries(killingProcessPromises).length;
 
-  const restartApp = async () => {
-    appState.willRestart = false;
-    appState.willQuit = false;
-    await createWindow();
-    await runEngineAll();
-  };
-
   // すべてのエンジンプロセスが停止している
   if (numLivingEngineProcess === 0) {
-    if (appState.willRestart) {
-      // 再起動フラグが立っている場合はフラグを戻して再起動する
-      log.info(
-        "All ENGINE processes killed. Now restarting app because of willRestart flag"
-      );
+    // 既にpostEngineKillが実行されている
+    if (appState.didPostEngineKillRun) {
+      if (appState.willRestart) {
+        // 再起動フラグが立っている場合はフラグを戻して再起動する
+        log.info(
+          "Post engine kill process done. Now restarting app because of willRestart flag"
+        );
+        event.preventDefault();
+
+        appState.willRestart = false;
+        appState.willQuit = false;
+        appState.didPostEngineKillRun = false;
+
+        createWindow().then(() => runEngineAll());
+      } else {
+        log.info("Post engine kill process done. Now quit app");
+      }
+    } else {
+      log.info("Running post engine kill process");
```

Comment:
ここ、必ず`handleEngineDirs()`を呼んで、それを待ってから`willRestart`の判定を入れると`preventDefault`を１回減らせそうに感じました！
こんな感じです：
```ts
// appState.didPostEngineKillRunは消去

log.info("Running post engine kill process");
await vvppManager.handleEngineDirs();

if (appState.willRestart) {
  // 再起動フラグが立っている場合はフラグを戻して再起動する
  log.info(
    "Post engine kill process done. Now restarting app because of willRestart flag"
  );
  event.preventDefault();

  appState.willRestart = false;
  appState.willQuit = false;
  appState.didPostEngineKillRun = false;

  createWindow().then(() => runEngineAll());
} else {
  log.info("Post engine kill process done. Now quit app");
}
```

Reply:
ここ、awaitできないんですよね。
かといってwillRestartの時はthen、終了時はPromiseにするとPromiseが実行される前に終わってしまうのでつらいところ。

Reply:
> ここ、awaitできないんですよね。

こうすればawaitできる･･････？
```ts
app.on("before-quit", async (event) => {
```

（`"before-quit"`が何度も呼ばれるのは複雑なので極力避けたいという思いがあります！）

Reply:
何か問題があるのかな、って思ってasyncにしてなかったんですが、そんなことなかったですね。

Reply:
（めっちゃ同感です。）